### ISSUE TITLE
A better bean injection strategy?

### ISSUE THREAD

**Issue by achuinard** [KuliginStepan/mongration#21]
Hi again - been using Mongration in my application and it works well. However, I have two MongoTemplate beans in my app, and in my `MongoChangelog` class, I'd like to use one MongoTemplate in some @Changeset methods and a different MongoTemplate in other @Changeset methods. I don't think your bean injection strategy supports this though:

```
protected Mono<Object> executeChangeSetMethod(Object changelog, Method changesetMethod) {
        Object[] parameterBeans = Arrays.stream(changesetMethod.getParameterTypes())
            .map(context::getBean)
            .toArray();
        try {
            return Mono.justOrEmpty(ReflectionUtils.invokeMethod(changesetMethod, changelog, parameterBeans));
        } catch (Exception e) {
            return Mono.error(e);
        }
    }
```

Is it possible you could inject beans into the method while also looking at their parameter name too? i.e. my two beans in my application are called `mongoTemplateFirstDb` and `mongoTemplateSecondDb` and it'd be nice to just put `MongoTemplate mongoTemplateSecondDb` as a method argument.

**Comment by KuliginStepan** [comment]
Hi! This is cool feature. I think we can support @Qualifier annotation as standard spring annotation for this purpose.

**Comment by achuinard** [comment]
That'd be great. Is it much of a change to support the annotation?

**Comment by KuliginStepan** [comment]
I need to do some research, but I think it would not be hard

**Comment by achuinard** [comment]
Hi just checking in. I can take a stab at this over the weekend myself if you'd like. Thanks.

**Comment by KuliginStepan** [comment]
Hi! I am going to implement this feature on this weekend.

**Comment by KuliginStepan** [comment]
@achuinard Hi! Support for `@Qualifier` available with v1.4.0.

**Comment by achuinard** [comment]
@KuliginStepan Much appreciated!
